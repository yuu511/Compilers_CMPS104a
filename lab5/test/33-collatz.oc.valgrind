==3842== Memcheck, a memory error detector
==3842== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3842== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==3842== Command: ../oc 33-collatz.oc
==3842== 
==3842== 
==3842== HEAP SUMMARY:
==3842==     in use at exit: 0 bytes in 0 blocks
==3842==   total heap usage: 709 allocs, 709 frees, 159,379 bytes allocated
==3842== 
==3842== All heap blocks were freed -- no leaks are possible
==3842== 
==3842== For counts of detected and suppressed errors, rerun with: -v
==3842== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
