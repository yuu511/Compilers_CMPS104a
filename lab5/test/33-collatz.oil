collatz:   .function int 
           .param int n
           .local int c
           c = 0  
.wh0:      goto .od0 if n == 1
.do0:      $t0:i = c + 1
           c = $t0:i  
.if0:      $t1:i = n % 2
           goto .el0 if $t1:i != 1
.th0:      $t3:i = 3 * n
           $t2:i = $t3:i + 1
           n = $t2:i  
.fi0       goto   
.el0:      $t4:i = n / 2
           n = $t4:i  
.fi0:
           goto .wh0  
.od0
           return c
           return
           .end
test:      .function 
           .param int n
           .local int c
           $t0:i = call collatz(n)  
           c = $t0:i  
           call putstr("collatz(")  
           call putint(n)  
           call putstr(") = ")  
           call putint(c)  
           call putchr('\n')  
           return
           .end
main:      .function int 
           call test(3)  
           call test(50)  
           return 0
           return
           .end
