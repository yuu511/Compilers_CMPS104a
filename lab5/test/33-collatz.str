string_set[   0]:    7296690267450279874 0x556ccf5d2cd8->"chr"
                   13348923631108115385 0x556ccf5d3698->"putstr"
string_set[   5]:   15118982290295364091 0x556ccf5d64a8->"test"
                   12679825168387489523 0x556ccf5d31f8->"num"
string_set[   6]:    9317282556992397945 0x556ccf5d5338->"if"
string_set[   7]:    2297668033614959926 0x556ccf5d4ad8->"0"
string_set[  10]:    6637313742931709005 0x556ccf5d5cd8->"*"
                    1583576014053361350 0x556ccf5d3858->"str"
                   13497578857207247412 0x556ccf5d2278->"string"
                   11597697714117577063 0x556ccf5d4a18->"="
string_set[  13]:   12868445110721718657 0x556ccf5d28f8->")"
string_set[  16]:   11911796782241914524 0x556ccf5d0fe8->"void"
                   10159970873491820195 0x556ccf5d4e58->"1"
string_set[  19]:    5344662657899890615 0x556ccf5d29b8->";"
string_set[  20]:   11868120863286193964 0x556ccf5d75f8->"main"
                   15816506932648033945 0x556ccf5d21b8->"("
string_set[  21]:    2617056156435215158 0x556ccf5d4d98->"!="
string_set[  25]:   11166701219533395393 0x556ccf5d3e78->"getstr"
string_set[  29]:   14635384904416252381 0x556ccf5d63c8->"return"
                    1370730858159036685 0x556ccf5d51b8->"+"
string_set[  31]:    6608372942718979652 0x556ccf5d54f8->"%"
string_set[  34]:    7211130878751977441 0x556ccf5d2b18->"putchr"
string_set[  35]:     821927901342812852 0x556ccf5d6c78->""collatz(""
                   14736352883882276958 0x556ccf5d5678->"=="
string_set[  37]:    6192799629375248351 0x556ccf5d73d8->"'\n'"
                   12291392672316406109 0x556ccf5d2818->"line"
string_set[  38]:   10462659240778295294 0x556ccf5d4478->"collatz"
string_set[  39]:   18413295556186398326 0x556ccf5d3038->"putint"
string_set[  40]:   10319632053171904911 0x556ccf5d4c58->"while"
string_set[  41]:    9826907452181669771 0x556ccf5d4638->"n"
string_set[  44]:      21606623943142546 0x556ccf5d4818->"{"
string_set[  45]:    6142509188972423790 0x556ccf5d23f8->""
string_set[  47]:   13686721134187162920 0x556ccf5d6388->"}"
string_set[  53]:   12137009137744407700 0x556ccf5d7ab8->"50"
string_set[  57]:    9386170080689500517 0x556ccf5d2338->"expr"
string_set[  58]:    4017852018407011111 0x556ccf5d25d8->"file"
                    3729804957429652673 0x556ccf5cbcc8->"<<ROOT>>"
string_set[  59]:   16579916199511856337 0x556ccf5d4178->"getln"
string_set[  61]:    4551451650890805270 0x556ccf5d55b8->"2"
string_set[  63]:    9611428142580219387 0x556ccf5d20b8->"fail"
                   10959529184379665549 0x556ccf5d4958->"c"
string_set[  64]:   12597418120160561823 0x556ccf5d2758->"int"
string_set[  69]:    1873634190020599143 0x556ccf5d5f78->"else"
string_set[  70]:    7419612306637001510 0x556ccf5d7028->"") = ""
                    9201800968376702500 0x556ccf5d3b78->"getchr"
string_set[  73]:   12865648880272446602 0x556ccf5d61b8->"/"
string_set[  74]:    8248777770799913213 0x556ccf5d5998->"3"
string_set[  78]:    4706206694200543916 0x556ccf5d20f8->","
load_factor = 0.608
bucket_count = 79
max_bucket_size = 4
