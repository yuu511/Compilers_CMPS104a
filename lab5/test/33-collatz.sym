main (7.28.1) {9} int function 

test (7.19.1) {8} void function 
   n (7.19.12) {9} int variable param lval 0
   c (7.20.4) {9} int variable local lval 1

collatz (7.9.1) {7} int function 
   n (7.9.14) {8} int variable param lval 0
   c (7.10.4) {8} int variable local lval 1

getln (6.30.1) {6} string function 

putchr (6.24.1) {1} void function 

getchr (6.28.1) {4} int function 

fail (6.22.1) {0} void function 

getstr (6.29.1) {5} string function 

putint (6.25.1) {2} void function 

putstr (6.26.1) {3} void function 

