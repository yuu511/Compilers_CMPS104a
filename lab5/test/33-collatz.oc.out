CONDITIONAL WHILE STATEMENT SYMBOL TOK_NE 
CONDITIONAL IF STATEMENT SYMBOL TOK_EQ 
